trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/weissinger_vortex_lattice.py

```diff
@@ -80,18 +82,88 @@ def weissinger_vortex_lattice(conditions,configuration,wing):
-        y  = np.atleast_2d(((i+1)*deltax-deltax/2))      
+        # Determine if wing segments are defined  
+        segment_keys = wing.Segments.keys()
+        n_segments   = len(segment_keys)
+        segment_vortex_index = np.zeros(n_segments)
+        # If spanwise stations are setup
+        if n_segments>0:
+            # discretizing the wing sections into panels
+            i              = np.arange(0,n)
+            j              = np.arange(0,n+1)
+            y_coordinates = (j)*deltax             
+            segment_chord = np.zeros(n_segments)
+            segment_twist = np.zeros(n_segments)
+            segment_sweep = np.zeros(n_segments)
+            segment_span = np.zeros(n_segments)
+            segment_chord_x_offset = np.zeros(n_segments)
+            section_stations = np.zeros(n_segments)
+            
+            # obtain chord and twist at the beginning/end of each segment
+            for i_seg in xrange(n_segments):                
+                segment_chord[i_seg] = wing.Segments[segment_keys[i_seg]].root_chord_percent*root_chord
+                segment_twist[i_seg] = wing.Segments[segment_keys[i_seg]].twist
+                segment_sweep[i_seg] = wing.Segments[segment_keys[i_seg]].sweeps.quarter_chord
+                section_stations[i_seg] = wing.Segments[segment_keys[i_seg]].percent_span_location*span
+                
+                if i_seg == 0:
+                    segment_span[i_seg] = 0.0
+                    segment_chord_x_offset[i_seg] = 0.25*root_chord # weissinger uses quarter chord as reference
+                else:
+                    segment_span[i_seg]    = wing.Segments[segment_keys[i_seg]].percent_span_location*span - wing.Segments[segment_keys[i_seg-1]].percent_span_location*span
+                    segment_chord_x_offset[i_seg]  = segment_chord_x_offset[i_seg-1] + segment_span[i_seg]*np.tan(segment_sweep[i_seg-1])
+            
+            # shift spanwise vortices onto section breaks 
+            for i_seg in xrange(n_segments):
+                idx =  (np.abs(y_coordinates-section_stations[i_seg])).argmin()
+                y_coordinates[idx] = section_stations[i_seg]
+            
+            # define y coordinates of horseshoe vortices      
+            ya = np.atleast_2d(y_coordinates[i])           
+            yb = np.atleast_2d(y_coordinates[i+1])          
+            deltax = y_coordinates[i+1] - y_coordinates[i]
+            xa =  np.zeros(n)
+            x  = np.zeros(n)
+            y  =  np.zeros(n)
+            twist_distri =  np.zeros(n)
+            section_length =  np.zeros(n)
+            
+            # define coordinates of horseshoe vortices and control points
+            i_seg = 0
+            for idx in xrange(n):
+                twist_distri[idx]   =  segment_twist[i_seg] + ((yb[0][idx] - deltax[idx]/2 - section_stations[i_seg]) * (segment_twist[i_seg+1] - segment_twist[i_seg])/segment_span[i_seg+1])     
```

Comment:
I don't think this behaves as you expect.

Why did you calculate the all of the section properties but only use the first segment?

I think you meant to put this as a nested for loop but that is also highly frowned upon in SUAVE.

Reply:
The twist distribution between each section is a  function of the starting segment twist plus a "delta twist" computer as a linear relationship between segments. As the span distance increases past a section, the segment number in increased (see line 141)